lib/client/mfa.go

```diff
@@ -99,14 +99,17 @@ func PromptMFAChallenge(
 		wg.Wait()
 	}
 
-	// Use otpCtx and otpCancel to cancel an ongoing OTP read.
+	// Use variables below to cancel OTP reads and make sure the goroutine exited.
+	otpWait := &sync.WaitGroup{}
```

Comment:
If you only have a single waiter, wouldn't a channel suffice?

Reply:
I tried that, but found the WaitGroup to be a bit cleaner in this case. That is because sometimes we have no OTP goroutine, so there are zero members to wait. With the channel we have to handle that explicitly.

Let me know if you prefer the channel anyway.